naga/src/front/wgsl/error.rs

```diff
@@ -1179,8 +1179,3 @@ impl<'a> Error<'a> {
         }
     }
 }
-
-#[test]
-fn test_error_size() {
-    assert!(size_of::<Error<'_>>() <= 48);
-}
```

Comment:
issue: I actually think we should test _each_ entry point where we return `Error`, and `assert!(…)` that it's pointer-sized now. If it's important to keep it small with `Box`, it's still important enough to test!

Reply:
The impact of large errors in the success case is that it affects performance, because `Result` return values need to be handled with `memcpy` instead of being returned in registers. Do you want us to have an assertion in every function that returns `naga::front::wgsl::Result`?

Reply:
Oh, by "entry point", do you mean, every Naga crate entry point to the WGSL front end?

None of the public entry points to the WGSL front end return `Error`. They all convert to a `ParseError` before returning.

Using the `front::wgsl::Result` type enforces the boxing of the error. This seems better than tests.

Or maybe I just don't understand what you mean?